src/librustdoc/html/length_limit.rs

```diff
@@ -0,0 +1,94 @@
+//! See [`HtmlWithLimit`].
+
+use std::fmt::Write;
+use std::ops::ControlFlow;
+
+use crate::html::escape::Escape;
+
+/// A buffer that allows generating HTML with a length limit.
+///
+/// This buffer ensures that:
+///
+/// * all tags are closed,
+/// * only the most recently opened tag is closed,
```

Comment:
Doesn't this sentence contradicts the previous one?

Reply:
Sorry, the wording might be a bit unclear. The first point means that all tags are closed *eventually*, once writing to the buffer is `finish`ed. To illustrate the second point, imagine we have this buffer:

```html
<p><em>Foo
```

If you call `close_tag`, it will close the most recently opened tag, `em`; it's not possible to close the `p` tag before closing the `em` tag, which would result in invalid HTML.

Perhaps it would be clearer if I rephrased it as follows?

```suggestion
/// * tags are closed in the reverse order of when they were opened (i.e., the correct HTML order),
```

Reply:
Yes it'd be better. :)

Also, this code reminds me of a code I wrote in rustdoc as well... Funny that we both write the same in different places. :)